services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/FeatureToggleAspect.java

```diff
@@ -17,10 +24,41 @@
 @Aspect
 @Component
 @Slf4j
-public class FeatureGateAspect {
+public class FeatureToggleAspect {
 
   @Autowired private TableFeatureToggle tableFeatureToggle;
 
+  @Autowired private TblPropsToggleRegistry tblPropsToggleRegistry;
+
+  @Around("@annotation(tblPropsEnabler)")
+  public boolean checkTblPropEnabled(
+      ProceedingJoinPoint proceedingJoinPoint, TblPropsEnabler tblPropsEnabler) throws Throwable {
+    if (((MethodSignature) proceedingJoinPoint.getSignature()).getReturnType() == boolean.class
+        && proceedingJoinPoint.getArgs()[1] instanceof TableDto
+        && proceedingJoinPoint.getArgs()[0] instanceof String) {
+      TableDto tableDto = (TableDto) proceedingJoinPoint.getArgs()[1];
+      String key = (String) proceedingJoinPoint.getArgs()[0];
+
+      Optional<String> feature = tblPropsToggleRegistry.obtainFeatureByKey(key);
+      if (!feature.isPresent()) {
+        return (boolean) proceedingJoinPoint.proceed();
+      }
+
+      boolean tableFeatureEnabled =
+          tableFeatureToggle.isFeatureActivated(
+              tableDto.getDatabaseId(), tableDto.getTableId(), feature.get());
+
+      // feature activation overwriting the decision of annotated method
+      return ((boolean) proceedingJoinPoint.proceed()) || tableFeatureEnabled;
```

Comment:
the annotated method is keypreserved. if feature  is `enable_mor`,  isKeypreserved should return false for `write.delete.mode` right? so that user can run alter table SQLs?  but here it will return `true` because `tableFeatureEnabled` is true. 
I may not be understanding this correctly

Reply:
Yeah I think you caught it right. I confused myself for the semantic of `preservedKey` here. 

- `isKeypreserved` by default return true for `write.delete.mode` 
- with feature of "enable_mor" active for a table `t`, `isKeypreserved` return false for `write.delete.mode`  
- so the condition should be ` ((boolean) proceedingJoinPoint.proceed()) && !tableFeatureEnabled;` for `enable_preservedKey` feature

Reply:
unfortunately all my unit tests got flipped 👎  great catch.